### PR TITLE
Enforce Network Keyword Consistency

### PR DESCRIPTION
**Pull request by bska**
The standard network model (`GRUPNET`) prohibits any extended network model keywords and vice versa.  Similarly, the `*PROP` keywords require the `NETWORK` keyword, and the `NODEPROP` keyword also requires the `BRANPROP` keyword.

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this please

File: tests/parser/NetworkTests.cpp [hunk 1]
```diff
@@ -68,6 +75,10 @@ BOOST_AUTO_TEST_CASE(Branch) {
 
 BOOST_AUTO_TEST_CASE(INVALID_DOWNTREE_NODE) {
```
Comment (atgeirr) [inline tests/parser/NetworkTests.cpp:76]: I am not sure what this and the next was supposed to test, but after the change won't both just throw due to both NETWORK and GRUPNET?

File: src/opm/input/eclipse/share/keywords/000_Eclipse100/N/NODEPROP [hunk 2]
```diff
@@ -3,6 +3,8 @@
   "sections": [
     "SCHEDULE"
   ],
+  "requires": ["NETWORK", "BRANPROP"],
```
Comment (atgeirr) [inline src/opm/input/eclipse/share/keywords/000_Eclipse100/N/NODEPROP:6]: I was not really aware of this feature, have we had it all the time?!?

**Review by atgeirr — APPROVED** [review]
Nice, should be merged.

File: tests/parser/NetworkTests.cpp (hunk 1 above)
Comment (bska) [reply tests/parser/NetworkTests.cpp:76]: > but after the change won't both just throw due to both `NETWORK` and `GRUPNET`?

I don't see `GRUPNET` here.  There is `GRUPTREE` but no `GRUPNET` as far as I can tell.

File: src/opm/input/eclipse/share/keywords/000_Eclipse100/N/NODEPROP (hunk 2 above)
Comment (bska) [reply src/opm/input/eclipse/share/keywords/000_Eclipse100/N/NODEPROP:6]: > have we had [`requires`/`prohibits`] all the time?!?

The feature was introduced in October 2020 (PR #1996), but it's not particularly widely deployed.  Most of its usage is in the table input code, and especially in the saturation functions (`SOF3` etc.) to reject decks using saturation function tables from both families.

**Review by bska — COMMENTED** [review]
Thanks a lot for your review.  I've tried to address your comments.

**Comment by bska** [comment]
PR approved and build check is green.  I'll merge into master.

File: tests/parser/NetworkTests.cpp (hunk 1 above)
Comment (atgeirr) [reply tests/parser/NetworkTests.cpp:76]: My mistake, thanks for pointing it out.

**Pull request merged**